v3/spring-cloud-aws-v3-test/src/test/java/io/awspring/cloud/test/v3/cloudwatch/CloudwatchMetricsIntegrationTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.awspring.cloud.test.v3.cloudwatch;
+
+import java.time.Duration;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.containers.localstack.LocalStackContainer.Service;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.given;
+
+@SpringBootTest(properties = { "management.metrics.export.cloudwatch.namespace=awspring/spring-cloud-aws" })
+@Testcontainers(disabledWithoutDocker = true)
+class CloudwatchMetricsIntegrationTest {
+
+	@Container
+	static LocalStackContainer localstack = new LocalStackContainer(
+			DockerImageName.parse("localstack/localstack:0.14.0")).withServices(Service.CLOUDWATCH);
+
+	@DynamicPropertySource
+	static void localstackProperties(DynamicPropertyRegistry registry) {
+		registry.add("spring.cloud.aws.credentials.access-key", localstack::getAccessKey);
+		registry.add("spring.cloud.aws.credentials.secret-key", localstack::getSecretKey);
+		registry.add("spring.cloud.aws.region.static", localstack::getRegion);
+		registry.add("management.metrics.export.cloudwatch.endpoint",
+				() -> localstack.getEndpointOverride(Service.CLOUDWATCH).toString());
+	}
+
+	@Autowired
+	private MeterRegistry meterRegistry;
+
+	@Test
+	void test() {
+		Counter testCounter = Counter.builder("test").tag("module", "cloudwatch").register(this.meterRegistry);
+		testCounter.increment();
+		given().await().atMost(Duration.ofSeconds(20))
```

Comment:
Correct me if I am wrong but I think `testCounter.count()` does not actually go to CloudWatch to check the value of this metric? If we have an integration test like that, ideally i believe it should verify that CloudWatch received this increment.

Reply:
that's right! I will submit changes involving request to aws cloudwatch instead.